packages/supertape/lib/supertape.d.ts

```diff
@@ -5,58 +5,326 @@ import {
+     */
+    deepEqual: <A, E>(actual: A, expected: E, message?: string) => EqualResult<A, E>;
+    
+    /**
+     * Asserts that `actual` and `expected` are not loosely equal, with different
+     * structure and/or nested values.
+     * 
+     * @note uses node's `deepEqual()` algorithm with strict comparisons
+     * (`===`) on leaf nodes
+     * 
+     * @param actual The resulting value to be tested.
+     * @param expected The value to be tested against.
+     * @param message An optional description of the assertion.
+     * 
+     * @since v3.3.0
+     */
+    notDeepEqual: <A, E>(actual: A, expected: E, message?: string) => EqualResult<A, E>;
+    
+    /**
+     * Asserts that `actual` is truthy.
+     * 
+     * @param actual The resulting value to be tested.
+     * 
+     * @since v3.1.0
+     */
+    ok: <A>(actual: boolean | A, message?: string) => OkResult<A, true>;
+    
+    /**
+     * Asserts that `actual` is falsy.
+     * 
+     * @param actual The resulting value to be tested.
+     * 
+     * @since v3.1.0
+     */
+    notOk: <A>(actual: boolean | A, message?: string) => OkResult<A | string, false>;
+    
+    /**
+     * Generates a passing assertion.
+     * 
+     * @param message An optional description of the assertion.
+     * 
+     * @since v3.2.0
+     */
+    pass: (message?: string) => PassResult;
+    
+    /**
+     * Generates a failing assertion.
+     * 
+     * @param message A description of the assertion.
+     * @param at 
+     * 
+     * @since v3.1.0
+     */
+    fail: (message: string, at?: string) => FailResult<string>;
+    
+    /**
+     * Declares the end of a test explicitly. `t.end()` must be
+     * called once (and only once) per test, and no further
+     * assertions are allowed.
+     * 
+     * @since v3.1.0
+     */
     end: () => void;
+    
+    /**
+     * Asserts that `actual` matches the regex `pattern`.
+     * 
+     * @note if `pattern` is not a valid regex, the assertion fails.
+     * 
+     * @param actual The resulting value to be tested.
+     * @param pattern A regex to be tested against.
+     * @param message An optional description of the assertion.
+     * 
+     * @since v5.1.0
+     */
+    match: (actual: string, pattern: string | RegExp, message?: string) => MatchResult | FailResult;
+    
+    /**
+     * Asserts that `actual` does not match the regex `pattern`.
+     * 
+     * @note if `pattern` is not a valid regex, the assertion always fails.
+     * 
+     * @param actual The resulting value to be tested.
+     * @param pattern A regex to be tested against.
+     * @param message An optional description of the assertion.
+     * 
+     * @since v5.6.0
+     */
+    notMatch: (actual: string, pattern: string | RegExp, message?: string) => MatchResult;
+};
+
+type CommentOperator = {
+    /**
+     * Prints a message without breaking the `tap` output.
+     * 
+     * @param message The message to be printed.
+     * 
+     * @since v3.1.0
```

Comment:
Also I don’t thin since has a lot sense here

Reply:
Maybe let's move types changes to other PR and add tests for it similar to [test/errors.ts](https://github.com/coderaiser/supertape/blob/v7.7.1/packages/supertape/test/errors.ts). 
What do you think?

Reply:
Okay! I'm not sure what the best way to do that is though. I'm not familiar enough with git to know how to make a branch with all of the current commits but only for `supertape.d.ts`. Should I just make a new branch and then revert all README commits?

I also don't really know how to test the type definitions.

Reply:
https://git-scm.com/docs/git-cherry-pick